src/dxgi/dxgi_swapchain_dispatcher.h

```diff
@@ -0,0 +1,280 @@
+#pragma once
+
+#include "dxgi_swapchain.h"
+
+namespace dxvk {
+
+  class DxgiSwapChainDispatcher : public IDXGISwapChain4 {
+
+  public:
+
+    DxgiSwapChainDispatcher(IDXGISwapChain4* dispatch)
+      : m_dispatch(dispatch) {
+    }
+
+    virtual ~DxgiSwapChainDispatcher() {
```

Comment:
This doesn't really need to be virtual since nothing will ever inherit from this class. We can just use the default destructor.

Reply:
Gives a warning if I don't do this.

Reply:
even if you mark the class as `final`?